tests/testthat/test-init.R

```diff
@@ -140,6 +140,32 @@ testthat::test_that("init data accepts a list of TealDatasetConnector object", {
   testthat::expect_no_error(init(data = dsc1, modules = modules(example_module())))
 })
 
+testthat::test_that("init data accepts teal_data_module", {
+  testthat::expect_no_error(
+    init(
+      data = teal_data_module(ui = function(id) div(), server = function(id) NULL),
+      modules = modules(teal:::example_module())
+    )
+  )
+})
+
+testthat::test_that("init teal_data_module doesn't accept ui and server with other formals than id", {
+  testthat::expect_error(
+    init(
+      data = teal_data_module(ui = function(id, x) div(), server = function(id) NULL),
+      modules = modules(teal:::example_module())
+    ),
+    "Must have exactly 1 formal arguments"
+  )
+  testthat::expect_error(
+    init(
+      data = teal_data_module(ui = function(id) div(), server = function(id, x) NULL),
+      modules = modules(teal:::example_module())
+    ),
+    "Must have exactly 1 formal arguments"
+  )
+})
```

Comment:
Arguments of module functions are checked by `teal_data_module`, is this not redundant? Unless there are no unit tests for `teal_data_module`...

Reply:
Good point. I'll push a change soon